ext/commands/subscribe.py

```diff
@@ -0,0 +1,239 @@
+import json
+import discord
+import asyncio
+import aiohttp
+from discord.ext import commands
+from ..infoscraper import FandomScrape, channelInfo
+from ..share.botUtils import chunks
+from ..share.dataGrab import getwebhook
+from ..share.prompts import subCheck, searchConfirm, searchPrompt
+
+async def subHolo(ctx: commands.Context, bot: commands.Bot):
+    listmsg = await ctx.send("Loading channels list...")
+
+    with open("data/channels.json", encoding="utf-8") as f:
+        channels = json.load(f)
+    csplit = []
+    for split in chunks(channels, 9):
+        csplit.append(split)
+    pagepos = 0
+    
+    while True:
+        picknum = 1
+        pickstr = ""
+        picklist = []
+        for split in csplit[pagepos]:
+            ytch = csplit[pagepos][split]
+            if ytch["category"] == "Hololive":
+                pickstr += f'{picknum}. {ytch["name"]}\n'
+                picklist.append(split)
+                picknum += 1
+        if pagepos == 0:
+            pickstr += f'\nA. Subscribe to all channels\nN. Go to next page\nX. Cancel'
+        elif pagepos == len(csplit) - 1:
+            pickstr += f'\nA. Subscribe to all channels\nB. Go to previous page\nX. Cancel'
+        else:
+            pickstr += f'\nA. Subscribe to all channels\nN. Go to next page\nB. Go to previous page\nX. Cancel'
+
+        listembed = discord.Embed(title="Subscribe to channel:", description=pickstr)
+        await listmsg.edit(content=None, embed=listembed)
+
+        def check(m):
+            return m.content.lower() in ['1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'n', 'b', 'x'] and m.author == ctx.author
+
+        while True:
+            try:
+                msg = await bot.wait_for('message', timeout=60.0, check=check)
+            except asyncio.TimeoutError:
+                await listmsg.delete()
+                await ctx.message.delete()
+                return
+            if msg.content in ['1', '2', '3', '4', '5', '6', '7', '8', '9']:
```

Comment:
CodeFactor somehow looks at this as an `elif` statement even though it's clearly an `if` statement. Ignoring.